test/controllers/tasks_controller_test.rb

```diff
@@ -3,124 +3,191 @@
     it "can get the new task page" do
-      skip
-
+      
+      
       # Act
       get new_task_path
-
+      
       # Assert
       must_respond_with :success
     end
   end
-
+  
   describe "create" do
     it "can create a new task" do
-      skip
-
+      
+      
       # Arrange
       task_hash = {
         task: {
           name: "new task",
           description: "new task description",
-          completion_date: nil,
+          completed: nil,
         },
       }
-
+      
       # Act-Assert
       expect {
         post tasks_path, params: task_hash
-      }.must_change "Task.count", 1
-
+      }.must_change 'Task.count', 1
+      
       new_task = Task.find_by(name: task_hash[:task][:name])
       expect(new_task.description).must_equal task_hash[:task][:description]
-      expect(new_task.due_date.to_time.to_i).must_equal task_hash[:task][:due_date].to_i
+      #expect(new_task.due_date.to_time.to_i).must_equal task_hash[:task][:due_date].to_i
       expect(new_task.completed).must_equal task_hash[:task][:completed]
-
       must_respond_with :redirect
       must_redirect_to task_path(new_task.id)
     end
   end
-
-  # Unskip and complete these tests for Wave 3
+  
+ # Tests for Wave 3
+  
   describe "edit" do
     it "can get the edit page for an existing task" do
-      skip
-      # Your code here
+      
+      #  Stepone Call on Edit#Action using path
+      #  Step Two: call on let(task) which is a local variable that will hold an created task. 
+      #  Step Three: What will the respond be...must_respond_with what? What do you expect. You expect it to return the task
+      get edit_task_path(task[:id])
+      
+      must_respond_with :success
+      
     end
-
+    
     it "will respond with redirect when attempting to edit a nonexistant task" do
-      skip
-      # Your code here
+      
+      get edit_task_path(-1)
+      
+      must_respond_with :redirect 
+      must_redirect_to tasks_path
     end
   end
-
+  
   # Uncomment and complete these tests for Wave 3
   describe "update" do
     # Note:  If there was a way to fail to save the changes to a task, that would be a great
     #        thing to test.
     it "can update an existing task" do
-      # Your code here
+      
+      #Arrange:
+      #step one get an existing book
+      #Step two once you have the existing book, change the name of the task. 
+      existing_task = task
+      updated_task_form_data = {
+        task: {
+          name: "cs fun",
+          description: "Recursion practice",
+          completed: Date.today
+        }
+      }
+      #Act:
+      #actually update the book data with the new hash form data you created.
+      patch task_path(existing_task.id), params: updated_task_form_data
+      
+      # Assert
+      #Step Three once you change the fields, check to see if they are different than original 
+      #expect(Book.find_by(id: existing_book.id).title).must_equal "Name that you changed book to"
```

Comment:
Not sure if this would help, but maybe if you can spend the time to remove these comments before submission, or find a way to reword them, that may help with solidfying your understanding